dbt/include/athena/macros/materializations/models/incremental/incremental.sql

```diff
@@ -11,10 +11,11 @@
   {% set force_batch = config.get('force_batch', False) | as_bool -%}
   {% set target_relation = this.incorporate(type='table') %}
   {% set existing_relation = load_relation(this) %}
-  {% set old_tmp_relation = adapter.get_relation(identifier=target_relation.identifier ~ '__dbt_tmp',
+  {% set tmp_table_suffix = adapter.generate_unique_temporary_table_suffix() %}
+  {% set old_tmp_relation = adapter.get_relation(identifier=target_relation.identifier ~ tmp_table_suffix,
```

Comment:
What if a user doesn't want to have a unique tmp table prefix? how about adding a new flag? `unique_tmp_table` that can be true or false. The default behaviour must be false, but in your case if using unique_tmp_table=true, you can then use benefit of the unique prefix.